flang/lib/Lower/OpenMP.cpp

```diff
@@ -178,14 +179,38 @@ genOMP(Fortran::lower::AbstractConverter &converter,
                          &clause.u)) {
         const Fortran::parser::OmpObjectList &ompObjectList = copyinClause->v;
         genObjectList(ompObjectList, converter, copyinClauseOperands);
+      } else if (const auto &allocateClause =
+                     std::get_if<Fortran::parser::OmpAllocateClause>(
+                         &clause.u)) {
+        mlir::Value allocator;
+        const Fortran::parser::OmpObjectList &ompObjectList =
+            std::get<Fortran::parser::OmpObjectList>(allocateClause->t);
+        int count = ompObjectList.v.size();
+        if (const auto &allocatorValue = std::get<
+                std::optional<Fortran::parser::OmpAllocateClause::Allocator>>(
+                allocateClause->t)) {
+          while (count--) {
+            allocator = fir::getBase(converter.genExprValue(
+                *Fortran::semantics::GetExpr(allocatorValue->v)));
+            allocatorOperands.push_back(allocator);
+          }
+        } else {
+          while (count--) {
+            auto val = firOpBuilder.getI32IntegerAttr(1);
+            allocator = firOpBuilder.createIntegerConstant(currentLocation,
```

Comment:
Can you reuse the same constant for each operand or do you really need to have the same constant created multiple times?

Reply:
I can reuse the same constant for the val here